README.md

```diff
@@ -1,2 +1,197 @@
+
+***`clientId`*** and ***`clientSecret`***
+
+Your Planship [authentication credentials](https://docs.planship.io/integration/#authentication-and-security). They can be also defined via the `PLANSHIP_API_CLIENT_ID` and `PLANSHIP_API_CLIENT_SECRET` variables.
+
+# Usage
+
+## Composables
+
+The `@planship/nuxt` module exports two composables implemented by the `@planship/vue` plugin: `usePlanshipCustomer` and `usePlanship`. The `@planship/vue` plugin is universal-mode friendly, meaning that these composables can be used for both server and client side rendering.
+
+### Working with entitlements and other customer data - `usePlanshipCustomer`
+
+In most rendering scenarios, your app will need to fetch and evaluate Planship entitlements for a specific customer. This can be accomplished with the Planship plugin and the `usePlanshipCustomer` function, which initializes a Planship API instance for a specific customer, continously fetches their `entitlements`, and exposes them in a reactive [Vue ref object](https://vuejs.org/api/reactivity-core.html#ref).
+
+The example below shows how customer entitlements are retrieved, and a simple boolean feature called `advanced-analytics` is used to conditionally render a `<NuxtLink>` element inside a Vue component.
+
+```vue
+<script setup>
+  import { usePlanshipCustomer } from '@planship/nuxt'
+
+  const { entitlements } = await usePlanshipCustomer('<CURRENT_CUSTOMER_ID>')
+</script>
+
+<template>
+  <NuxtLink
+    v-if="entitlements['advanced-analytics']"
+    to="/analytics"
+  >
+    Analytics
+  </NuxtLink>
+</template>
+```
+
+When `usePlanshipCustomer` is used on the client-side, entitlements are automatically updated via a WebSocket connection every time they change.
+
+When used in the universal rendering mode, the data is fetched in the following fashion:
+1. The plugin is initialized on the server and Planship entitlements and subscription data are fetched so they can be used for server-side rendering.
+2. The plugin is initialized on the client using the data already fetched on the server.
+3. THe pluging re-hydrates itslef on the client, and initiates a websocket connection for continuous entitlements updates from Planship.
+
+#### Composite return value for both `sync` and `async` operations
+
+The `usePlanshipCustomer` function returns a composite result that is both a promise and a data object the promise resolves to. This means that the function can be called both as a synchronous function or an asynchronous one using `await` (or `then/catch` chain).
+
+If you want to block code execution until customer entitlements are fetched from the Planship API, call the function with `await`:
+
+```ts
+const { entitlements } = await usePlanshipCustomer('<CURRENT_CUSTOMER_ID>')
+```
+
+If you want to return immediately and let entitlements be fetched asynchronously, call `usePlanshipCustomer` as a synchronous function:
```

Comment:
and let entitlements be fetched asynchronously -> and fetch entitlements asynchronously

call `usePlanshipCustomer` as a synchronous function -> simply call `usePlanshipCustomer` without `await`.